lib/jxl/enc_frame.cc

```diff
@@ -1016,9 +1016,6 @@ Status EncodeFrame(const CompressParams& cparams_orig,
   passes_enc_state->special_frames.clear();
 
   CompressParams cparams = cparams_orig;
-  if (ib.color_transform != ColorTransform::kNone) {
```

Comment:
Basically the idea is that if the input image is not RGB but it is already something else (YCbCr or XYB), then we need to override the color_transform choice in `cparams` to match what we already have.
The y4m input relies on this.

The problem with this is that it is done too late - at the frame level, while it should have been at the image level, because it is too late here to change xyb_encoded. This is not an issue for y4m or RGB input, but would be a problem if e.g. the original cparams says kNone and the input imagebundle is (somehow) in XYB already.

Reply:
Do you know of a case that reproduces the issue where the value in cparams doesn't match the input image's YCbCr or XYB? I could try to move this to, for example, enc_file.cc, but would be good to have a way to check that it works

Reply:
`ffmpeg -i some_image.png input.y4m`

Then try `cjxl input.y4m -q 100`.

Reply:
That worked (with -q 100 indeed, without it doesn't but that is an unrelated issue). I verified that without overriding color transform, the colors after decoding look wrong, and that moving this assignment to enc_file, fixes that, and also fixes fuzzer_corpus.